packages/studio-ui/src/components/AddPageButton/LayoutSelector.tsx

```diff
@@ -0,0 +1,68 @@
+import { ChangeEvent, useCallback, useState } from "react";
+import { FlowStepModalProps } from "./FlowStep";
+import useStudioStore from "../../store/useStudioStore";
+import DialogModal from "../common/DialogModal";
+import { selectCssClasses } from "../UnionPropInput";
+import { twMerge } from "tailwind-merge";
+
+export default function LayoutSelector({
+  isOpen,
+  handleClose,
+  handleConfirm,
+}: FlowStepModalProps) {
+  const layouts = useStudioStore(
+    (store) => store.layouts.layoutNameToLayoutState
+  );
+  const [selectedLayout, setSelectedLayout] = useState<string>("");
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const onConfirm = useCallback(async () => {
+    const layoutState = layouts[selectedLayout];
+
+    try {
+      await handleConfirm(layoutState);
+      await handleClose();
+    } catch (err: unknown) {
+      if (err instanceof Error) {
+        setErrorMessage(err.message);
+      } else {
+        throw err;
+      }
+    }
+  }, [handleClose, handleConfirm, layouts, selectedLayout]);
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      setSelectedLayout(e.target.value);
+    },
+    [setSelectedLayout]
+  );
+
+  const selectClasses = twMerge(selectCssClasses, "w-full mb-6");
+  const body = (
+    <label>
+      <select className={selectClasses} onChange={handleChange}>
+        <option key="" value="">
+          No layout selected
+        </option>
+        {Object.entries(layouts).map(([layoutName]) => (
```

Comment:
If you're only using the `layoutName` in the `map`, could we just do `Object.keys(layouts)` here?